src/extensions/data_array.js

```diff
@@ -67,24 +96,101 @@ export default () => {
       });
     };
 
+    const createInstances = observationObjects => {
+      return new Promise(resolve => {
+        db().then(models => {
+          Promise.all(observationObjects.map(observationObj => {
+            const newReq = Object.create(req);
+            newReq.body = observationObj.body;
+            return models.createInstance(observations, newReq, exclude).then(
+              instance => {
+              observationObj.instanceId = instance.id;
+              return Promise.resolve();
```

Comment:
That was my initial approach. However, I discovered the array of instances Ids returned by ```Promise.all``` were not necessarily in the same order as the Observation information in the request , which I assumed was happening due to the instances being created asynchronously -- and causes a problem for the URL ordering requirement of the response. I wondered if the instance Ids would always be created in ascending order based on the order of the ```createInstance``` calls and if so would provide a solution -- sort the array of instance Ids -- but I abandoned that thinking in favor of the approach where the instance id is assigned to the ``observationObj``` since then I could guarantee I would have the instance Ids in the same order as the input.

Is it guaranteed the instance Ids will be created in ascending order based on the order of the ```createInstance``` calls? Or is there a better approach I haven't thought of?